### ISSUE TITLE
Mocking request / response improvement

### ISSUE THREAD

**Issue by michal-studniarski-orba** [selenide/selenide#1254]
As we discuss on gitter i would like to improve request / response intercepting on this way that if Configuration.proxy = true then below options available:

* `selenide.start()` - to run server
* `selenide.stop()` - to turn off
* `stubResponseFor(String endpointUrl, int status,String responseFileNameFromResourcesDirectory)` - to stub response
* `stubRequestFor(String endpointUrl,String method,String requestWithHeadersAndBodyFileNameFromResourcesDirectory)` - to stub request

so on test it should looks like below

```java
@Test
public void testMethod() {
   selenide.start()
   stubResponseFor(with params)
   goToHomePage();
   clickOnSomething();
   assertSomething();
   selenide.stop
} 
```

**Comment by SeleniumTestAB** [comment]
I would also like to include such approach for the DevTools wrapping.

**Comment by pavelpp** [comment]
You can easily implement this functionality yourself by using proxy filters. We use it for mocking and intercepting network requests in our project.

**Comment by asolntsev** [comment]
@pavelpp Could you please share some link or code sample?

**Comment by pavelpp** [comment]
Implement your own RequestFilter and add it to the proxy with relevant parameters. At the core, you need only a couple of methods. (The example is in Kotlin)

```
	fun mockResponseFilter(exactUrl: String, response: String, statusCode: Int, method: String): RequestFilter {
		return RequestFilter { request: HttpRequest, _: HttpMessageContents?, messageInfo: HttpMessageInfo ->
			val url = messageInfo.originalUrl
			if (enabled && url == exactUrl && request.method().name() == method) {
				return@RequestFilter mockedHttpResponse(request, statusCode, response)
			}else{
				null
			}
		}
	}

	private fun mockedHttpResponse(request: HttpRequest, statusCode: Int, response: String): FullHttpResponse {
		val httpResponse: FullHttpResponse = DefaultFullHttpResponse(
			request.protocolVersion(), HttpResponseStatus(statusCode, "mocked by test"),
			Unpooled.wrappedBuffer(response.toByteArray())
		)
		log(request, response)
		httpResponse.headers()[HttpHeaders.CONTENT_LENGTH] = httpResponse.content().readableBytes()
		return httpResponse
	}
```

**Comment by asolntsev** [comment]
@michal-studniarski-orba @pavelpp This feature was released today in Selenide 6.9.0
